lib/util.js

```diff
@@ -66,6 +66,43 @@ function urlGenerate(aParsedUrl) {
 }
 exports.urlGenerate = urlGenerate;
 
+const MAX_CACHED_INPUTS = 32;
+
+/**
+ * Takes some function `f(input) -> result` and returns a memoized version of
+ * `f`.
+ *
+ * We keep at most `MAX_CACHED_INPUTS` memoized results of `f` alive. The
+ * memoization is a dumb-simple, linear least-recently-used cache.
+ */
+function lruMemoize(f) {
+  const cache = [];
+
+  return function (input) {
+    for (var i = 0; i < cache.length; i++) {
```

Comment:
Is this faster than `Array.indexOf`?

Reply:
We can't use `indexOf` here because the data in the array is an `Object` and so will be tested for pointer equality.

We could use `Array.prototype.find`, but now we're introducing functions and things, and the whole point of this code is to gain whatever control we can over execution, which higher-order-functions can be a bit anathema to.